### DISCUSSION TITLE
markdown: Sanitize HTML

### DISCUSSION THREAD

**Discussion by easyaspi314** [gsantner/markor#1615] [Ideas] [answered]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**. Not a question or feature request.
- [X] The issue is **not** already reported at [Issues](https://github.com/gsantner/markor/issues?q=) or [discussions](https://github.com/gsantner/markor/discussions?discussions_q=). _(I've searched it)_.
- [X] Markor **is** up to date. See [Releases](https://github.com/gsantner/markor/tags) for the latest version. Updates are available from [F-Droid](https://f-droid.org/en/packages/net.gsantner.markor/) and GitHub.

### Description

Currently, Markdown files are unrestricted in the HTML tags and attributes used. This allows for opening URIs/URLs and executing arbitrary JavaScript without direct interaction, which is usually unwanted behavior unless it is being used to build a website.

Most Markdown viewers will strip all but a small subset of HTML tags and attributes, preventing this issue.

This could be toggled in the settings for the few MD pages that use it.

### Steps to reproduce

1. Create a new markdown file
2. Paste one of the following POCs:
```html
<!-- script -->
<script>window.location.href="vnd.youtube:dQw4w9WgXcQ"</script>
```
```html
<!-- iframe -->
<iframe src="vnd.youtube:dQw4w9WgXcQ">
```
```html
<!-- on* attributes -->
<body onpageshow="window.location.href='vnd.youtube:dQw4w9WgXcQ'">
```
3. Click View
4. Get Rickroll'd. :trollface: 

### Information

Android version: 12
Device: Google Pixel 4a
App Version: 2.8.6

### Source

F-Droid

### Format / File type

Markdown

**Comment by gsantner** [comment, marked as answer]
Hello,

at Markor it is wanted/expected behaviour. Specifically, there are even options to globally inject **any** custom javascript and css. And..I know users do add custom js and css to some files, so the logic/style only applies specifically to these files in addition to globals. By the way, theres more than just Markdown at Markor, you can also open normal .html files just fine "as is".

> Most Markdown viewers

As you say, Most, but not all. It is especially bad to see that most sanitizers also break non-malicious things. One painful recent example - wanted to use NextCloud PicoCMS - and many of the things don't work. Simple things. Like displaying small webp favicon icon inline.

If you are not happy about that, Markor might be not what you look for. Don't really want to limit users in what they can do.

**Comment by easyaspi314** [comment]
Making the option toggleable is not limiting users at all.

**Reply by gsantner** [reply]
@easyaspi314 

feel free to spend time on it and develop it. If you add it, should be opt-in.  
Also, don't add loads of dependencies. Every dependency has it's cost and risk, and requires maintenance.
